config/src/grt.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright 2023-, GraphOps and Semiotic Labs.
+// SPDX-License-Identifier: Apache-2.0
+
+use bigdecimal::{BigDecimal, ToPrimitive};
+use serde::{de::Error, Deserialize};
+
+#[derive(Debug, PartialEq)]
+pub struct NonZeroGRT(u128);
+
+impl NonZeroGRT {
+    pub fn get_value(&self) -> u128 {
+        self.0
+    }
+}
+
+impl<'de> Deserialize<'de> for NonZeroGRT {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let v = BigDecimal::deserialize(deserializer)?;
+        if v <= 0.into() {
+            return Err(Error::custom("GRT value must be greater than 0"));
+        }
```

Comment:
Perhaps the "non-zero" check can be moved to the config validation? Though having it here reduces the amount of code for sure.